### PR TITLE
feat(nn): add configurable norm_layer to Mlp

### LINKED ISSUE #1451: 🚀[FEA]: Make `Mlp` layer normalization type configurable
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Low (would be nice)

### Please provide a clear description of problem you would like to solve.

Currently, `Mlp` is hardcoded to use `BatchNorm` as the only type of normalization. It would be great to extend this to other norms, like `LayerNorm` (and possibly accept user-supplied norms, like those from TransformerEngine).

Stems from discussion here with @coreyjadams @laserkelvin:
https://github.com/NVIDIA/physicsnemo/pull/1401#discussion_r2850009189

### Describe any alternatives you have considered

_No response_

### PR DESCRIPTION
**Pull request by Thabhelo**
# PhysicsNeMo Pull Request

## Description

Adds a `norm_layer` parameter to `physicsnemo.nn.Mlp` so normalization type is configurable instead of being limited to `BatchNorm1d` via `use_batchnorm`.

Supported options:
- `norm_layer="batchnorm"`, `"layernorm"`, or `"te_layernorm"`
- `norm_layer=nn.LayerNorm` or `get_layer_norm_class()` for TE-aware LayerNorm
- Any callable factory invoked as `norm_layer(out_features)`

`use_batchnorm=True` remains backward compatible and is mutually exclusive with `norm_layer`.

Closes #1451

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [x] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation). *(N/A: `physicsnemo.nn` layer change, not `physicsnemo.models`.)*

## Notes

- `norm_layer` is documented in the `Mlp` class docstring and doctest examples; Sphinx exposes it via `docs/api/nn/layers/fully_connected_mlp.rst` automodule.

## Dependencies

None.

## Test plan

- [x] `pytest test/nn/module/test_mlp_layers.py -v`
- [x] `pytest --doctest-modules physicsnemo/nn/module/mlp_layers.py`
- [x] pre-commit hooks on changed files (ruff, interrogate, markdownlint, license, import-linter)

### CODE REVIEW

**Comment by Thabhelo** [comment]
Okay cool, I have addressed the feedback in the latest push:

- `"layernorm"` now maps to PyTorch `nn.LayerNorm`; `"te_layernorm"` requires `transformer_engine` and CUDA (raises otherwise).
- `get_layer_norm_class()` remains available as a user-supplied factory for TE-aware auto selection.
- `self.norm_layer` is stored on `Mlp` for introspection.

Added tests for the TE string path and the stored attribute, too.

File: physicsnemo/nn/module/mlp_layers.py [hunk 1]
```diff
@@ -14,25 +14,66 @@
...
+        )
+
+    if norm_layer is not None:
+        if isinstance(norm_layer, str):
+            key = norm_layer.lower().replace("-", "_")
+            if key in {"batchnorm", "batch_norm", "bn"}:
+                return nn.BatchNorm1d
+            if key in {"layernorm", "layer_norm", "ln", "te_layernorm"}:
+                layer_norm = get_layer_norm_class()
+                return layer_norm
```
[the remark this answers is not in the document]
Comment (Thabhelo) [reply physicsnemo/nn/module/mlp_layers.py]: Fixed in 8896888. `layernorm` now maps to PyTorch `nn.LayerNorm`, and `te_layernorm` requires transformer_engine plus CUDA.

File: physicsnemo/nn/module/mlp_layers.py [hunk 2]
```diff
@@ -108,12 +166,14 @@ def __init__(
...
         bias: bool = True,
         use_batchnorm: bool = False,
+        norm_layer: NormLayerSpec = None,
         spectral_norm: bool = False,
         use_te: bool = False,
     ):
         super().__init__()
 
         self.use_te = use_te
+        norm_factory = _resolve_norm_factory(norm_layer, use_batchnorm)
```
[the remark this answers is not in the document]
Comment (Thabhelo) [reply physicsnemo/nn/module/mlp_layers.py:192]: Fixed in 8896888. `self.norm_layer` is stored on the module now.

**Comment by Thabhelo** [comment]
Greptile feedback from the first review is addressed in 8896888. Replies are on the inline threads. docs come from the updated Mlp docstrings via Sphinx automodule.

**Comment by Thabhelo** [comment]
@loliverhennigh gentle ping when you have a moment. Greptile feedback is addressed in 8896888. I will rebase onto main shortly to clear the merge conflict. Thanks!
